File: i2f-secure/src/main/java/i2f/springboot/secure/consts/SecureConsts.java
package i2f.springboot.secure.consts;

/**
 * @author Ice2Faith
 * @date 2023/6/13 8:51
 * @desc
 */
public interface SecureConsts {
    int RSA_KEY_SIZE_1024 = 1024;
    int RSA_KEY_SIZE_2048 = 2048;

    int AES_KEY_SIZE_128 = 128;
    int AES_KEY_SIZE_192 = 192;
    int AES_KEY_SIZE_256 = 256;

    // secure sign web header
    String DEFAULT_SECURE_HEADER_NAME = "sswh";

    // secure sign web parameter
    String DEFAULT_SECURE_PARAMETER_NAME = "sswp";

    String FILTER_EXCEPTION_ATTR_KEY = "SECURE_FILTER_EXCEPTION";

    String FILTER_DECRYPT_HEADER = "SECURE_FILTER_DECRYPT";

    String FILTER_ENCRYPT_HEADER = "SECURE_FILTER_ENCRYPT";

    String FLAG_ENABLE = "true";

    String FLAG_DISABLE = "false";

    String DEFAULT_HEADER_SEPARATOR = ";";

    String SECURE_HEADER_ATTR_KEY = "SECURE_HEADER";


    // 是否是String返回值类型标记
    String STRING_RETURN_HEADER = "SECURE_RETURN_STRING";

    String ACCESS_CONTROL_EXPOSE_HEADERS = "Access-Control-Expose-Headers";

    String RSA_KEY_FILE_NAME = "rsa.key";

    // 是否包含动态刷新的RSA公钥签名/公钥头
    String SECURE_DYNAMIC_KEY_HEADER = "skey";

    String SECURE_REQUIRE_RESPONSE = "SECURE_REQUIRE_RESPONSE";

    String DEFAULT_ENC_URL_PATH = "/enc/";

}

File: i2f-secure/web/secure-web/src/secure/consts/secure-consts.js
/**
 * 定义常量
*/
const SecureConsts={
    // 默认安全头
    DEFAULT_SECURE_HEADER_NAME:()=>"sswh",

    DEFAULT_SECURE_PARAMETER_NAME:()=>"sswp",

    // 安全头的分隔符
    DEFAULT_HEADER_SEPARATOR:()=>";",

    // 动态RSA公钥的响应头
    SECURE_DYNAMIC_KEY_HEADER:()=>"skey",

    // 安全Body加密标记头
    SECURE_DATA_HEADER:()=>"secure-data",

    // 安全Param加密标记头
    SECURE_PARAMS_HEADER:()=>"secure-param",

    // 安全URL编码标记头
    SECURE_URL_HEADER:()=>"secure-url",

    // 启用标记位
    FLAG_ENABLE:()=>"true",

    // 禁用标记位
    FLAG_DISABLE:()=>"false",

    // 默认的URL编码请求路径
    ENC_URL_PATH:()=>"/enc/",
}

export default SecureConsts